src/jaxsim/api/contact.py

```diff
@@ -395,3 +406,96 @@ def mixed_jacobian(W_H_C: jtp.Matrix, W_J_WC: jtp.Matrix) -> jtp.Matrix:
             raise ValueError(output_vel_repr)
 
     return O_J_WC
+
+
+@jax_dataclasses.pytree_dataclass
+class ContactsState(JaxsimDataclass, abc.ABC):
+    """
+    Abstract class storing the state of the contacts model.
+    """
+
+    @classmethod
+    def build(cls, **kwargs) -> ContactsState:
+        """
+        Build the contact state object.
+        Returns:
+            The contact state object.
+        """
+
+        return cls(**kwargs)
+
+    @classmethod
+    def zero(cls, **kwargs) -> ContactsState:
+        """
+        Build a zero contact state.
+        Returns:
+            The zero contact state.
+        """
+
+        return cls.build(**kwargs)
+
+    def valid(self, **kwargs) -> bool:
+        """
+        Check if the contacts state is valid.
+        """
+
+        return True
+
+
+@jax_dataclasses.pytree_dataclass
+class ContactsParams(JaxsimDataclass, abc.ABC):
+    """
+    Abstract class representing the parameters of a contact model.
+    """
+
+    @abc.abstractmethod
+    def build(self) -> ContactsParams:
+        """
+        Create a `ContactsParams` instance with specified parameters.
+        Returns:
+            The `ContactsParams` instance.
+        """
+
+        raise NotImplementedError
+
+    def valid(self, *args, **kwargs) -> bool:
+        """
+        Check if the parameters are valid.
+        Returns:
+            True if the parameters are valid, False otherwise.
+        """
+
+        return True
+
+
+@jax_dataclasses.pytree_dataclass
+class ContactModel(abc.ABC):
+    """
+    Abstract class representing a contact model.
+    Attributes:
+        parameters: The parameters of the contact model.
+        terrain: The terrain model.
+    """
+
+    parameters: ContactsParams = dataclasses.field(default_factory=ContactsParams)
+    terrain: jaxsim.terrain.Terrain = dataclasses.field(
+        default_factory=jaxsim.terrain.FlatTerrain
+    )
+
+    @abc.abstractmethod
+    def contact_model(
+        self,
+        position: jtp.Vector,
+        velocity: jtp.Vector,
+        **kwargs,
+    ) -> tuple[jtp.Vector, jtp.Vector]:
```

Comment:
If we abstract the soft contacts class, I'd take advantage for renaming this method as follows:

```suggestion
    @abc.abstractmethod
    def compute_contact_forces(
        self,
        position: jtp.Vector,
        velocity: jtp.Vector,
        **kwargs,
    ) -> tuple[jtp.Vector, tuple[Any, ...]]:
```

Usually, more advanced contact models, operate in different stages. We might need to introduce new methods associated to those in stages in the future for alternative contact models.

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/178/commits/d8ef45331383e92253bb310441b0afea1f662c81